res/controllers/Traktor-Kontrol-S2-MK1-hid-scripts.js

```diff
@@ -0,0 +1,1361 @@
+var ButtonBrightnessOff = 0x00;
+var ButtonBrightnessOn = 0x1F;
+
+// KNOWN ISSUES:
+// * The effect button LEDs briefly flicker when pressing the effect focus button.
+
+// eslint definitions
+var TraktorS2MK1 = new function() {
+    this.controller = new HIDController();
+
+    // When true, packets will not be sent to the controller.
+    // Used when updating multiple LEDs simultaneously.
+    this.batchingLEDUpdate = false;
+
+    // Previous values, used for calculating deltas for encoder knobs.
+    this.previousBrowse = 0;
+    this.previousPregain = {
+        "[Channel1]": 0,
+        "[Channel2]": 0
+    };
+    this.previousLeftEncoder = {
+        "[Channel1]": 0,
+        "[Channel2]": 0
+    };
+    this.previousRightEncoder = {
+        "[Channel1]": 0,
+        "[Channel2]": 0
+    };
+    this.wheelTouchInertiaTimer = {
+        "[Channel1]": 0,
+        "[Channel2]": 0
+    };
+
+    this.topEncoderPressed = {
+        "[Channel1]": false,
+        "[Channel2]": false
+    };
+    this.leftEncoderPressed = {
+        "[Channel1]": false,
+        "[Channel2]": false
+    };
+    this.shiftPressed = {
+        "[Channel1]": false,
+        "[Channel2]": false
+    };
+
+    this.padModes = {
+        "hotcue": 0,
+        "introOutro": 1,
+        "sampler": 2
+    };
+    this.currentPadMode = {
+        "[Channel1]": this.padModes.hotcue,
+        "[Channel2]": this.padModes.hotcue
+    };
+    this.padConnections = {
+        "[Channel1]": [],
+        "[Channel2]": []
+    };
+
+    this.lastTickValue = [0, 0];
+    this.lastTickTime = [0.0, 0.0];
+    this.syncEnabledTime = {};
+
+    this.longPressTimeoutMilliseconds = 275;
+
+    this.effectButtonLongPressTimer = {
+        "[EffectRack1_EffectUnit1]": [0, 0, 0, 0],
+        "[EffectRack1_EffectUnit2]": [0, 0, 0, 0]
+    };
+    this.effectButtonIsLongPressed = {
+        "[EffectRack1_EffectUnit1]": [false, false, false, false],
+        "[EffectRack1_EffectUnit2]": [false, false, false, false]
+    };
+    this.effectFocusLongPressTimer = {
+        "[EffectRack1_EffectUnit1]": 0,
+        "[EffectRack1_EffectUnit2]": 0
+    };
+    this.effectFocusChooseModeActive = {
+        "[EffectRack1_EffectUnit1]": false,
+        "[EffectRack1_EffectUnit2]": false
+    };
+    this.effectFocusButtonPressedWhenParametersHidden = {
+        "[EffectRack1_EffectUnit1]": false,
+        "[EffectRack1_EffectUnit2]": false
+    };
+    this.previouslyFocusedEffect = {
+        "[EffectRack1_EffectUnit1]": null,
+        "[EffectRack1_EffectUnit2]": null
+    };
+    this.effectButtonLEDconnections = {
+        "[EffectRack1_EffectUnit1]": [],
+        "[EffectRack1_EffectUnit2]": []
+    };
+};
+
+TraktorS2MK1.registerInputPackets = function() {
+    var MessageShort = new HIDPacket("shortmessage", 0x01, this.shortMessageCallback);
+    var MessageLong = new HIDPacket("longmessage", 0x02, this.longMessageCallback);
```

Comment:
I think it will make the core less readable. Just one character would be different between `InputReport0x01` and `InputReport0x02`. Maybe the name should indicate what is sent and not what number it has (just search for the name and you will find the number).

Reply:
Why bother with a name field in `HIDPacket` if it can be auto derived from the other parameters?

Reply:
The whole common-hid-packet-parser.js is legacy code, which needs a complete rewrite from scratch. For backward compatibility, we keep the old syntax as it is.
But the code inside common-hid-packet-parser.js is modernized and documented in the 2.4 branch, where we have a newer JavaScript engine which supports the ES6 language standard.

Reply:
So does it even make sense to merge this mapping if it uses legacy stuff?

Reply:
Are you happy with the renaming I have done?

Reply:
Yes, rewriting common-hid-packet-parser.js is a long term task - not expected for 2023.

Reply:
So is there already a new interface?

Reply:
No